tests/providers/test_python.py

```diff
@@ -3,13 +3,46 @@
 import unittest
 import warnings
 
+from typing import Any, Dict, Iterable, Optional, Union
 from unittest.mock import patch
 
 import pytest
 
 from faker import Faker
 
 
+@pytest.mark.parametrize(
+    "object_type, kwargs",
+    (
+        (None, {}),
+        (bool, {}),
+        (str, {}),
+        (float, {}),
+        (int, {}),
+        (tuple, {}),
+        (set, {}),
+        (list, {}),
+        (Iterable, {}),
+        (dict, {}),
+    ),
+)
+def test_pyobject(
+    object_type: Optional[Union[bool, str, float, int, tuple, set, list, Iterable, dict]],
```

Comment:
If we define a `TypeVar`, we can import it and use it here

Reply:
Same as [here](https://github.com/joke2k/faker/pull/1822#discussion_r1135813843)